src/hotspot/share/gc/shenandoah/shenandoahOldGeneration.cpp

```diff
@@ -386,29 +391,34 @@ bool ShenandoahOldGeneration::validate_transition(State new_state) {
   ShenandoahHeap* heap = ShenandoahHeap::heap();
```

Comment:
Could you update the ascii art of the sate transition diagram to include WAITING_FOR* states instead of the old WAITING state?